Vic2ToHoI4/Source/HOI4World/Regions/Regions.cpp

```diff
@@ -80,4 +80,27 @@ std::vector<std::string> HoI4::Regions::getSuperregions(const std::string& regio
 	}
 
 	return superregions;
+}
+
+
+bool HoI4::Regions::isRegionBlocked(const std::string& stateRegion, const std::string& ownerRegion) const
+{
+	if (stateRegion == ownerRegion)
+	{
+		return true;
+	}
+
+	const auto& blockedRegions = getBlockedRegions(ownerRegion);
+	if (std::find(blockedRegions.begin(), blockedRegions.end(), stateRegion) != blockedRegions.end())
```

Comment:
I feel like std::any_of would be more clear about the intent, though it might end up harder to read.

Reply:
Maybe something like this, to improve readability?
`    struct DivisibleBy
    {
        const int d;
        DivisibleBy(int n) : d(n) {}
        bool operator()(int n) const { return n % d == 0; }
    };
 
    if (std::any_of(v.cbegin(), v.cend(), DivisibleBy(7))) {
        std::cout << "At least one number is divisible by 7\n";
    }`

Reply:
I think
```
if (std::any_of(blockedRegions.cbegin(), blockedRegions.cend(),[stateRegion](const auto& region){return region == stateRegion}))
```

Though you can always move the labda to another line and assign it if it looks better.
```
auto regionIsBlocked =  [stateRegion](const auto& region){return region == stateRegion};
if ((std::any_of(blockedRegions.cbegin(), blockedRegions.cend(), regionIsBlocked)
```

Reply:
Perfect